client/templates/courses/tagged-courses.js

```diff
@@ -1,8 +1,49 @@
+Template.taggedCourses.onCreated(function(){
+  // Get reference to template instance
+  var instance = this;
+
+  // Accessing the Iron.controller to invoke getParams method of Iron Router.
+  var router = Router.current();
+
+  // Getting Params of the URL
+  instance.tag = router.params.tag;
+
+  // Subscribe to courses tagged with the current tag
+  instance.subscribe('taggedCourses', instance.tag);
+
+  // Subscribe to course images
+  instance.subscribe('images');
+});
+
+
+Template.taggedCourses.rendered = function () {
+  // Get reference to template instance
+  var instance = this;
+
+  // Set the page site title for SEO
+  Meta.setTitle('Courses tagged "' + instance.tag + '"');
+};
+
 Template.taggedCourses.helpers({
     'courses': function () {
-        return Courses.find().fetch();
+      // Get reference to template instance
+      var instance = Template.instance();
+
+      // Get tag from template instance
+      var tag = instance.tag;
+
+      // Fetch courses matching current tag
+      var taggedCourses = Courses.find({"keywords": tag}).fetch();
```

Comment:
I added a "keywords" filter to the query, so that it will only return courses where keywords contains tags. This ensures that the page will show only tagged courses, despite other active subscriptions.